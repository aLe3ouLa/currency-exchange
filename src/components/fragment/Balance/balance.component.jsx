import React from "react";

import { BalanceContainer } from "./balance.styles";
import { getCurrencySymbol } from "../../../utils/currencies";

const Balance = ({ balance, currency, fromValue, isFromValue }) => {
  const currentBalance = balance[currency] ? balance[currency] : 0;
  const currentCurrency = getCurrencySymbol(currency);

  const errorBalance = balance[currency] < fromValue;
  return (
    <BalanceContainer error={isFromValue && errorBalance}>
      Balance: {currentCurrency} {currentBalance}
    </BalanceContainer>
  );
};

export default Balance;
