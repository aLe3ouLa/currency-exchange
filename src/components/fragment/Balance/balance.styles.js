import styled, { css } from "styled-components";

export const BalanceContainer = styled.p`
  font-size: 1.2rem;
  margin-bottom: calc(
    ${({ theme }) => theme.spacing.sm} + ${({ theme }) => theme.spacing.xxs}
  );
  ${({ error }) =>
    error
      ? css`
          color: ${({ theme }) => theme.color.error};
        `
      : css`
          color: ${({ theme }) => theme.color.gray};
        `};
`;
