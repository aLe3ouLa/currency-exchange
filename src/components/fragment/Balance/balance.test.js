import React from "react";
import ReactDOM from "react-dom";
import Balance from "./balance.component";
import { cleanup, render } from "@testing-library/react";

import { ThemeProvider } from "styled-components";
import { theme, Defaults } from "../../../styles";

describe("Balance tests", () => {
  afterEach(cleanup);

  const props = {
    balance: {
      EUR: 5,
    },
    currency: "EUR",
  };

  it("renders the Balance component without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <Defaults />
          <Balance {...props} />
        </React.Fragment>
      </ThemeProvider>,
      div
    );
  });

  it("has a valid snapshot ", () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <Defaults />
          <Balance {...props} />
        </React.Fragment>
      </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct output text for props ", () => {
    const { getByText } = render(
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <Defaults />
          <Balance {...props} />
        </React.Fragment>
      </ThemeProvider>
    );

    expect(getByText("Balance: â‚¬ 5")).toBeInTheDocument();
  });
});
