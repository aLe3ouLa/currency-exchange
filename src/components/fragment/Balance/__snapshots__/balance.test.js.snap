// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balance tests has a valid snapshot  1`] = `
<DocumentFragment>
  <p
    class="sc-AxjAm gtFmXl"
  >
    Balance: â‚¬ 5
  </p>
</DocumentFragment>
`;
